### PR TITLE
define env type

### PR DESCRIPTION
**Pull request by holic**
fixes https://github.com/sindresorhus/execa/issues/1132

I've seen the issue described in #1132 happen in both Remix and Next.js projects

### CODE REVIEW

**Comment by ehmicky** [comment]
Hi @holic,

Thanks for this PR.

The issue at #1132 was specific to Remix. It is due to a bug in Remix currently tracked in https://github.com/remix-run/remix/issues/9718. It impacts `node:child_process`, and is not specific to Execa. So there's nothing to fix in Execa.

When it comes to Next.js, could you please show an example of a type error when using both Execa and Next.js?

**Comment by ehmicky** [comment]
So basically the core problem is this: the following pattern is bad.

```ts
// Augment `ProcessEnv` type from Node.js
interface ProcessEnv { 
  NODE_ENV: "development" | "production" | "test"; 
} 
```

The intent is to change the type of `process.env.NODE_ENV`. However, `ProcessEnv` is also used by Node.js to type the `env` option of `node:child_process`. So the above creates unintentional problems.

Unfortunately, this pattern was introduced by Next.js, which was seemingly copied by Remix, and might have been copied by other libraries.

Although this is not a bug with Execa, and since getting that bug fixed in either Next.js or Remix seems difficult, I would suggest for Execa to patch that typing problem.

From that perspective, your solution looks good. However, could you please add some type tests? Specifically, a type test that does the `interface Process { ... }` augmentation, then calling Execa with the `env` option, and make sure the types work in that case. Thanks!

**Comment by holic** [comment]
I agree the original pattern and source of this issue isn't great (haven't we learned enough that fiddling with globals is a bad idea?) but seems easy to simplify types here while maintaining intended use and compatibility.

Happy to add some tests! Will do in the next day or so.

**Comment by holic** [comment]
Added! Thanks for making that an easy copy+paste job for me :)

File: types/arguments/options.d.ts [hunk 1]
```diff
@@ -77,7 +76,7 @@ export type CommonOptions<IsSync extends boolean = boolean> = {
 
 	@default [process.env](https://nodejs.org/api/process.html#processenv)
 	*/
-	readonly env?: typeof env;
+	readonly env?: Partial<Record<string, string>>;
```
Comment (ehmicky) [inline types/arguments/options.d.ts:79]:
```suggestion
	readonly env?: Readonly<Partial<Record<string, string>>>;
```

**Review by ehmicky — APPROVED** [review]
Looks good to me, except the one small suggestion above.

To be more explicit, the previous type of `options.env` was:

```ts
{ 
  TZ?: string;
  [key: string]: string | undefined;
}
```

And the new type is:

```ts
{ 
  readonly [key: string]: string | undefined;
}
```

This is backward compatible with the former one, i.e. this is not a breaking change.

@sindresorhus What do you think?

**Comment by sindresorhus** [comment]
👍

**Comment by ehmicky** [comment]
Thanks @holic!

**Pull request merged**